### PR TITLE
[linux] install kmps to user directory by default in gui

### PR DESCRIPTION
**Pull request by glasseyes**
Fixes: #1177
install kmps to user directory by default in gui

add `-s` option to `km-package-install` to install to shared area i.e. `/usr/local/share/keyman`.
similarly for `km-package-uninstall` and `km-package-list-installed`
in the shared install, install all the files to `/usr/local/share/keyman/<kbid>` instead of going via a temp directory. This completes #1189. Then hard link the relevant files to the doc and font directories

let `km-package-list-installed` and `km-config` know about system installed kmps in `/usr/share/keyman`

---
This change is [Reviewable](https://reviewable.io/reviews/keymanapp/keyman/1208)

### CODE REVIEW

File: linux/keyman-config/keyman_config/get_kmp.py [hunk 1]
```diff
@@ -44,6 +43,15 @@ def get_download_folder():
 	"""
 	Folder where downloaded files will be saved.
 
+	Returns:
+	    str: path of user keyman cache folder
+	"""
```
Comment (darcywong00) [inline linux/keyman-config/keyman_config/get_kmp.py]: Maybe this description (compared to `keyman_cache_dir()`)?
```
str: path where downloaded files will be saved
```

File: linux/keyman-config/keyman_config/get_kmp.py [hunk 2]
```diff
@@ -44,6 +43,15 @@ def get_download_folder():
...
+	return keyman_cache_dir()
+
+def keyman_cache_dir():
+	"""
+	User keyman cache folder
+
 	Returns:
 	    str: path of user keyman cache folder
```
Comment (darcywong00) [inline linux/keyman-config/keyman_config/get_kmp.py]: Could also document the cache folder will be created if it doesn't exist

File: linux/keyman-config/keyman_config/install_kmp.py [hunk 3]
```diff
@@ -65,143 +69,218 @@ def get_infdata(tmpdirname):
...
+	# just get latest version of kmn unless there turns out to be a way to get the version of a file at a date
+	base_url = "https://raw.github.com/keymanapp/keyboards/master/" + sourcePath
+	kmn_url = base_url + "/source/" + kbid + ".kmn"
+	response = requests.get(kmn_url)
+	if response.status_code == 200:
+		kmndownloadfile = os.path.join(kbdir, kbid + ".kmn")
+		with open(kmndownloadfile, 'wb') as f:
+			f.write(response.content)
+			logging.info("Installing %s.kmn as keyman file", kbid)
+		kmn_file = os.path.join(kbdir, kbid + ".kmn")
```
Comment (darcywong00) [inline linux/keyman-config/keyman_config/install_kmp.py]: Isn't this also
`kmn_file = kmndownloadfile`?

File: linux/keyman-config/keyman_config/get_kmp.py (hunk 1 above)
Comment (glasseyes) [reply linux/keyman-config/keyman_config/get_kmp.py]: good point, done in next commit

File: linux/keyman-config/keyman_config/get_kmp.py (hunk 2 above)
Comment (glasseyes) [reply linux/keyman-config/keyman_config/get_kmp.py]: good point, done in next commit

File: linux/keyman-config/keyman_config/install_kmp.py (hunk 3 above)
Comment (glasseyes) [reply linux/keyman-config/keyman_config/install_kmp.py]: good point, done in next commit

**Pull request merged**